### PR TITLE
example: update telegram translation bot

### PR DESCRIPTION
**Pull request by localvar**

### CODE REVIEW

File: doc/cookbook/translation-bot.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+As we have already explained the basic idea behind the bot, it is easy to see the whole process from the flow above. However, because the final response needs to combine the execution results of multiple APIs, we need to use multiple namespaces to store the parameters and execution results of these APIs, i.e., the requests sent and the responses they return.
+
+And, to achieve a better effect, we also defined some data on the pipeline:
+
+```yaml
+data:
+  zh:
+    fallback: "(抱歉，我不懂这种语言。)"
+    text:  "中文🇨 🇳 "
```
Comment (haoel) [inline doc/cookbook/translation-bot.md:137]:
```suggestion
    text:  "中文🇨🇳"
```

File: doc/cookbook/translation-bot.md [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+  ja:
+    fallback: "(申し訳ないのですが、この言葉は本当に初めて見ました。)"
+    text: "やまと🇯 🇵 "
```
Comment (haoel) [inline doc/cookbook/translation-bot.md:140]:
```suggestion
    text: "やまと🇯🇵"
```

File: doc/cookbook/translation-bot.md [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+  en:
+    fallback: "(I'm sorry, but I really don't know this language.)"
+    text:  "English🇬 🇧 "
```
Comment (haoel) [inline doc/cookbook/translation-bot.md:143]:
```suggestion
    text:  "English🇬🇧"
```

File: doc/cookbook/translation-bot.md [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+    result1: processVoice            # voice
+    result2: processPhoto            # photo
+    "": END                          # ignore, end the processing
+ 
+# text message
+- filter: requestBuilderExtractText
+  alias: processText                 # alias of the filter
+  namespace: extract                 # namespace the filter belongs to
+  jumpIf:                            # conditional jump, begin translation
+    "": translateChinese             # if everything is fine, or end the
```
Comment (haoel) [inline doc/cookbook/translation-bot.md:57]: why do use the empty key?

File: doc/cookbook/translation-bot.md [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+
+Easegress is the next-generation traffic-based gateway product developed by MegaEase. It is completely architected on top of cloud-native technology, avoiding the shortcomings of traditional reverse proxy in terms of high availability, traffic orchestration, monitoring, service discovery, etc.
+
+We released Easegress v2.0 recently, with another significant enhancement to traffic orchestration, allowing users to implement a super API by orchestrating multiple APIs without writing any code. This article will demonstrate this feature by building a Telegram translation bot. This bot can automatically translate incoming messages into Chinese, Japanese, and English, and, in addition to text messages, it also supports translating voice and photo messages.
+
+## 1. Prerequisites
+
+Since the bot needs to receive Telegram message notifications and call third-party APIs, we must prepare the following in advance:
+
+* Install the latest version of Easegress according to [this document](https://github.com/megaease/easegress#setting-up-easegress) and make sure that external applications can access the Easegress instance on at least one of ports 80, 88, 443 or 8443.
```
Comment (haoel) [inline doc/cookbook/translation-bot.md:11]:
```suggestion
* Install the latest version of Easegress according to [this document](https://github.com/megaease/easegress#setting-up-easegress) and make sure that external applications can access the Easegress instance on at least one of ports 80, 88, 443, or 8443.
```

File: doc/cookbook/translation-bot.md [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+
+The sequence chart below shows the workflow of this bot:
+
+![sequence chart](../imgs/translation-bot-sequence-chart.png)
+
+Upon receiving a notification of a new message from the Telegram server via webhook, the bot first checks the message type and does the following accordingly：
+
+* **Text Message**: Extract message text directly；
+* **Voice Message**: In this case, the message body only contains the ID of the voice file, so we need to call Telegram's API to convert the ID to a file address, then download the file and send its contents to Google's voice recognition service to convert it to text；
+* **Photo Message**: This case is basically the same as the voice message, but the file content is sent to Google's Image Annotation service.
```
Comment (haoel) [inline doc/cookbook/translation-bot.md:28]:
```suggestion
* **Photo Message**: Basically, this case is the same as the voice message, but the file content is sent to Google's Image Annotation service.
```

**Review by haoel — APPROVED** [review]

File: doc/cookbook/translation-bot.md (hunk 4 above)
Comment (localvar) [reply doc/cookbook/translation-bot.md:57]: because it needs to jump if everything is fine, to skip `processVoice` and `processPhoto`.

**Review by suchen-sci — APPROVED** [review]

**Pull request merged**